packages/token-bridge-sdk/src/hooks/arbTokenBridge.types.ts

```diff
@@ -47,25 +47,116 @@ export enum AssetType {
   ETH = 'ETH'
 }
 
-export type TransactionLifecycle<Tx, TxReceipt> = Partial<{
-  onTxSubmit: (tx: Tx) => void
-  onTxConfirm: (txReceipt: TxReceipt) => void
-}>
+export enum TokenSymbol {
+  ETH = 'ETH'
+}
 
-export type L1EthDepositTransactionLifecycle = TransactionLifecycle<
-  L1EthDepositTransaction,
-  L1EthDepositTransactionReceipt
->
+export type EthDepositTransactionLifecycle = {
+  onL1TxSubmit: ({ tx }: { tx: L1EthDepositTransaction }) => void
+  onL1TxConfirm: ({
```

Comment:
Let's be consistent and have `on*TxSuccess` everywhere (right now we have success / confirm). We should also have failure (even if it's not used at the moment)